Assets/Scripts/Settings/TimeSettings.cs

```diff
@@ -0,0 +1,34 @@
+﻿
+using Assets.Scripts.Gamemode;
+using System;
+
+namespace Assets.Scripts.Settings
+{
+    public class TimeSettings
+    {
+       
+        public float? currentTime { get; set; }
+        public float? dayLength { get; set; }
+        public bool? pause { get; set; }
```

Comment:
Current time and pause do not have a default setting, thus there may be a null reference exception.

Reply:
Currently set from the prefabs variables, of which maybe map settings are better, will check that.